spec/controllers/article_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe ArticlesController, type: :request do
+  SAMPLE_URL = "www.test-img-url.com/blabla"
+  let(:article) { build(:article) }
+  let(:valid_response_double) { instance_double(RestClient::Response,
+                                                body: {
+                                                  'results' => [{ 'urls' => { 'raw' => SAMPLE_URL } }]
+                                                }.to_json) }
+  let(:invalid_response_double) { instance_double(RestClient::Response,
+                                                  body: {
+                                                    'total' => 0
+                                                  }.to_json) }
+
+  def create_call(article_attributes, response)
+    allow(RestClient).to receive(:get).and_return(response)
+    post "/articles", params: { article: article_attributes }
+  end
+
+  describe 'POST create' do
+
+    context 'when is successful' do
+      before(:each) { create_call article.attributes, valid_response_double }
+
+      it 'redirects' do
+        expect(response).to have_http_status(:found)
+      end
+
+      it 'renders image from image API' do
+        follow_redirect!
+        expect(response.body).to match(".*src=#{SAMPLE_URL}.*")
+      end
+    end
+
+    context 'when data is wrong' do
+      before { create_call({ :title => nil }, invalid_response_double) }
```

Comment:
Always try to use new syntax for hashes
```suggestion
      before { create_call({ title: nil }, invalid_response_double) }
```